components/google-cloud/google_cloud_pipeline_components/experimental/custom_job.py

```diff
@@ -0,0 +1,223 @@
+            dsl_utils.resolve_cmd_lines(
+                component_op.component_spec.implementation.container.command,
+                _is_output_parameter
+            )
+            worker_pool_spec['containerSpec'][
+                'command'
+            ] = component_op.component_spec.implementation.container.command
+
+        if component_op.component_spec.implementation.container.args:
+            dsl_utils.resolve_cmd_lines(
+                component_op.component_spec.implementation.container.args,
+                _is_output_parameter
+            )
+            worker_pool_spec['containerSpec'][
+                'args'
+            ] = component_op.component_spec.implementation.container.args
+        if accelerator_type is not None:
+            worker_pool_spec['machineSpec']['acceleratorType'
+                                           ] = accelerator_type
+        if accelerator_count is not None:
+            worker_pool_spec['machineSpec']['acceleratorCount'
+                                           ] = accelerator_count
+        if boot_disk_type is not None:
+            if 'diskSpec' not in worker_pool_spec:
+                worker_pool_spec['diskSpec'] = {}
+            worker_pool_spec['diskSpec']['bootDiskType'] = boot_disk_type
+        if boot_disk_size_gb is not None:
+            if 'diskSpec' not in worker_pool_spec:
+                worker_pool_spec['diskSpec'] = {}
+            worker_pool_spec['diskSpec']['bootDiskSizeGb'] = boot_disk_size_gb
+
+        job_spec['workerPoolSpecs'] = [worker_pool_spec]
+        if replica_count is not None and replica_count > 1:
+            additional_worker_pool_spec = copy.deepcopy(worker_pool_spec)
+            additional_worker_pool_spec['replicaCount'] = str(replica_count - 1)
+            job_spec['workerPoolSpecs'].append(additional_worker_pool_spec)
+
+    if timeout is not None:
+        if 'scheduling' not in job_spec:
+            job_spec['scheduling'] = {}
+        job_spec['scheduling']['timeout'] = timeout
+    if restart_job_on_worker_restart is not None:
+        if 'scheduling' not in job_spec:
+            job_spec['scheduling'] = {}
+        job_spec['scheduling']['restartJobOnWorkerRestart'
+                              ] = restart_job_on_worker_restart
+    if service_account is not None:
+        job_spec['serviceAccount'] = service_account
+    if network is not None:
+        job_spec['network'] = network
+
+    custom_job_payload = {
+        'displayName': display_name or component_op.component_spec.name,
+        'jobSpec': job_spec
+    }
+
+    custom_job_component_spec = ComponentSpec(
+        name=component_op.component_spec.name,
+        inputs=component_op.component_spec.inputs + [
+            InputSpec(name='gcp_project', type='String'),
+            InputSpec(name='gcp_region', type='String')
+        ],
+        outputs=component_op.component_spec.outputs +[OutputSpec(
```

Comment:
Nit: Perhaps format the outputs same as inputs (you can add training comma after `OutputSpec(...)`) for that.

Reply:
Done, change both to use extend.